test.sh

```diff
@@ -268,28 +266,70 @@ if [ x"$package_registry" == x"true" ]; then
     fi
 fi
 
-docker run --rm \
-    -v ${aws_credentials_file}:/root/.aws/credentials:ro \
-    -e AWS_PROFILE="${aws_profile}" \
-    -e DCOS_ENTERPRISE="$enterprise" \
-    -e DCOS_LOGIN_USERNAME="$DCOS_LOGIN_USERNAME" \
-    -e DCOS_LOGIN_PASSWORD="$DCOS_LOGIN_PASSWORD" \
-    -e CLUSTER_URL="$CLUSTER_URL" \
-    -e S3_BUCKET="$S3_BUCKET" \
-    ${azure_args} \
-    -e SECURITY="$security" \
-    -e PYTEST_ARGS="$PYTEST_ARGS" \
-    ${FRAMEWORK_ARGS} \
-    -e STUB_UNIVERSE_URL="$STUB_UNIVERSE_URL" \
-    -e PACKAGE_REGISTRY_ENABLED="${package_registry}" \
-    -e PACKAGE_REGISTRY_STUB_URL="${PACKAGE_REGISTRY_STUB_URL}" \
-    -e DCOS_FILES_PATH="${dcos_files_path}" \
-    -v "${dcos_files_path}":"${dcos_files_path}" \
-    -v $(pwd):$WORK_DIR \
-    -v $ssh_path:/ssh/key \
-    -w $WORK_DIR \
-    -t \
-    ${DOCKER_INTERACTIVE_FLAGS} \
-    ${DOCKER_ARGS} \
-    mesosphere/dcos-commons:latest \
-    ${DOCKER_COMMAND}
+if [ -n "$dcos_files_path" ]; then
+    dcos_files_volume_arg="-v \"${dcos_files_path}\":\"${dcos_files_path}\""
+fi
+
+if [ -n "$TEAMCITY_VERSION" ]; then
+    # The teamcity python module treats present-but-empty as enabled.
+    # We must therefore completely omitted this envvar to disable teamcity handling.
+    echo "TEAMCITY_VERSION=\"${TEAMCITY_VERSION}\"" >> $envfile
+fi
+
+if [ -n "$azure_enabled" ]; then
+    cat >> $envfile <<EOF
+AZURE_CLIENT_ID=$AZURE_DEV_CLIENT_ID
+AZURE_CLIENT_SECRET=$AZURE_DEV_CLIENT_SECRET
+AZURE_TENANT_ID=$AZURE_DEV_TENANT_ID
+AZURE_STORAGE_ACCOUNT=$AZURE_DEV_STORAGE_ACCOUNT
+AZURE_STORAGE_KEY=$AZURE_DEV_STORAGE_KEY
+EOF
+fi
+
+cat >> $envfile <<EOF
+AWS_PROFILE=$aws_profile
+CLUSTER_URL=$CLUSTER_URL
+DCOS_ENTERPRISE=$enterprise
+DCOS_FILES_PATH=$dcos_files_path
+DCOS_LOGIN_PASSWORD=$DCOS_LOGIN_PASSWORD
+DCOS_LOGIN_USERNAME=$DCOS_LOGIN_USERNAME
+FRAMEWORK=$framework
+PACKAGE_REGISTRY_ENABLED=$package_registry
+PACKAGE_REGISTRY_STUB_URL=$PACKAGE_REGISTRY_STUB_URL
+PYTEST_ARGS=$PYTEST_ARGS
+S3_BUCKET=$S3_BUCKET
+SECURITY=$security
+STUB_UNIVERSE_URL=$STUB_UNIVERSE_URL
+EOF
+
+while read line; do
+    # Prefix match, then strip prefix in envfile:
+    if [[ "${line:0:8}" = "TEST_SH_" ]]; then
+        echo ${line#TEST_SH_} >> $envfile
+    fi
+done < <(env)
+
+CMD="docker run --rm \
+${docker_interactive_arg} \
```

Comment:
`-t` and `-i` are very often specified together. Would it make sense to include them here together?

Reply:
Moved `-t` up to here.